pkg/auth/google/auth.go

```diff
@@ -0,0 +1,67 @@
+package google
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/oneconcern/datamon/pkg/model"
+	goauth "google.golang.org/api/oauth2/v2"
+	goption "google.golang.org/api/option"
+)
+
+const timeout = 60 * time.Second
+
+// New returns a new instance of google Auth
+func New() Auth {
+	return Auth{}
+}
+
+// Auth implements Authable for google credentials
+type Auth struct {
+}
+
+// Principal queries google oauth2 with some local credentials to extract user
+// information (aka principal).
+//
+// By default, credentials are taken from your default application_default_credentials.
+// On linux, this is located at ~/.config/gcloud/application_default_credentials.json.
+func (g Auth) Principal(credFile string) (model.Contributor, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	svc, err := goauth.NewService(ctx,
+		goption.WithCredentialsFile(credFile),
+		goption.WithScopes(goauth.UserinfoEmailScope, goauth.UserinfoProfileScope),
+	)
+	if err != nil {
+		return model.Contributor{}, fmt.Errorf("invalid credentials: %w", err)
```

Comment:
```invalid credential``` => ```failed to use credentials```? It might be a network issue being reported as invalid creds.

Reply:
ok will change this

Reply:
=> forthcoming PR